src/ClimaLSM.jl

```diff
@@ -1,329 +1,121 @@
-    else
-        values = map((x) -> similar(state), keys)
-        return ClimaCore.Fields.FieldVector(;
-            model.model_name => (; zip(keys, values)...),
-        )
-    end
-end
-
-"""
-    initialize(model::AbstractModel)
-
-Creates the prognostic and auxiliary states structures, but with unset values
-not set; constructs and returns the coordinates for the `model` domain.
-We may need to consider this default more as we add diverse components and 
-`Simulations`.
-"""
-function initialize(model::AbstractModel)
-    coords = coordinates(model)
-    Y = initialize_prognostic(model, coords)
-    p = initialize_auxiliary(model, coords)
-    return Y, p, coords
-end
-
-initialize(model::NotIncluded{FT}) where {FT} = (;model.model_name => FT[]), (;model.model_name => FT[]), (;model.model_name => FT[])
-
-Domains.coordinates(model::AbstractModel) = Domains.coordinates(model.domain)
-
-
-
-
-
-
-#### LandModel Specific
+include("models.jl")
 include("Soil.jl")
 using .Soil
 include("Roots.jl")
 using .Roots
+
+export RootSoilModel
+
+
+abstract type AbstractLandModel{FT} <: AbstractModel{FT} end
 """
-    struct LandModel{FT, SM <: AbstractModel{FT}, RM <: AbstractModel{FT}} <: AbstractModel{FT}
+    struct RootSoilModel{FT, SM <: AbstractModel{FT}, RM <: AbstractModel{FT}} <: AbstractLandModel{FT}
 
 A concrete type of `AbstractModel` for use in land surface modeling. Each component model of the
 `LandModel` is itself an `AbstractModel`.
 
 If a user wants to run in standalone, would they use this interface?
 No, but should it work ?
 """
-struct LandModel{
+struct RootSoilModel{
     FT,
     CT <: AbstractConfiguration{FT},
-    SM <: AbstractModel{FT},
-    VM <: AbstractModel{FT},
-    RM <: AbstractModel{FT},
-    SNM <: AbstractModel{FT},
-    CM <: AbstractModel{FT},
-} <: AbstractModel{FT}
+    SM <: AbstractSoilModel{FT},
+    VM <: AbstractVegetationModel{FT},
+} <: AbstractLandModel{FT}
     configuration::CT
     soil::SM
     vegetation::VM
-    river::RM
-    snow::SNM
-    carbon::CM
 end
 
-function LandModel{FT}(;
+function RootSoilModel{FT}(;
     soil_args::NamedTuple = (;),
     vegetation_args::NamedTuple = (;),
-    river_args::NamedTuple = (;),
-    snow_args::NamedTuple = (;),
-    carbon_args::NamedTuple = (;),
 ) where {FT}
-    
-    configuration = RootSoilConfiguration{FT}() 
-    soil =
-        isempty(soil_args) ? NotIncluded{FT}(:soil) :
-        Soil.RichardsModel{FT}(; configuration = configuration, soil_args...)
-    vegetation =
-        isempty(vegetation_args) ? NotIncluded{FT}(:vegetation) :
-        Roots.RootsModel{FT}(;
-            configuration = configuration,
-            vegetation_args...,
-        )
-    river = NotIncluded{FT}(:river)
-    snow = NotIncluded{FT}(:snow)
-    carbon = NotIncluded{FT}(:carbon)
-    args = (configuration, soil, vegetation, river, snow, carbon)
-    return LandModel{FT, typeof.(args)...}(args...)
+    #somehow need to pass in soil version and veg version...
```

Comment:
Right... perhaps adding args for the type of soil & root model to the constructor? Then you would call something like `RootSoilModel{FT}(RichardsModel, soil_args, RootsModel, vegetation_args)`. I think that may be simpler than trying to put the model version into `soil_args` itself.